eessi/testsuite/eessi_mixin.py

```diff
@@ -165,3 +175,46 @@ def assign_tasks_per_compute_unit(self):
     def request_mem(self):
         """Call hook to request the required amount of memory per node"""
         hooks.req_memory_per_node(self, app_mem_req=self.required_mem_per_node())
+
+    @run_after('setup')
+    def log_runtime_info(self):
+        """Log additional runtime information: which CVMFS repo was used (or if it was testing local software),
+        path to the modulefile, EESSI software subdir, EESSI testsuite version"""
+        self.postrun_cmds.append('echo "EESSI_CVMFS_REPO: $EESSI_CVMFS_REPO"')
+        self.postrun_cmds.append('echo "EESSI_SOFTWARE_SUBDIR: $EESSI_SOFTWARE_SUBDIR"')
+        if self.module_name:
+            # Get full modulepath, stripping leading spaces with sed and trailing colon by string substitution
+            get_full_modpath = f'modpath=$(module show {self.module_name} 2>&1 | '
+            get_full_modpath += 'grep ".lua" | '
+            get_full_modpath += 'sed "s/^[[:space:]]*//") && echo "FULL_MODULEPATH: ${modpath%:*}"'
+            self.postrun_cmds.append(get_full_modpath)
+
+    @run_after('run')
+    def extract_runtime_info_from_log(self):
+        """Extracts the printed runtime info from the job log and logs it as reframe variables"""
+        # If EESSI_CVMFS_REPO environment variable was set, extract it and store it in self.cvmfs_repo_name
+        # Try block is needed to surpress sanity error if there is no match
+        try:
+            repo_name = sn.extractsingle(r'EESSI_CVMFS_REPO: /cvmfs/(?P<repo>.*)$', f'{self.stagedir}/{self.stdout}',
+                                         'repo', str)
+            if repo_name:
+                self.cvmfs_repo_name = f'{repo_name}'
+        except SanityError:
+            pass
```

Comment:
i'd like to make a distinction between `'None'`, indicating a failing sanity check, and an empty string, indicating that the EESSI CVMFS repo was not used:
```suggestion
        try:
            self.cvmfs_repo_name = sn.extractsingle(r'EESSI_CVMFS_REPO: /cvmfs/(?P<repo>.*)$', f'{self.stagedir}/{self.stdout}',
                                         'repo', str)
        except SanityError:
            pass
```

Reply:
I'm... confused. If `EESSI_CVMFS_REPO` is empty, it will not match the pattern `EESSI_CVMFS_REPO: /cvmfs/(?P<repo>.*)$` either. I.e. both an unset `EESSI_CVMFS_REPO` and an empty `EESSI_CVMFS_REPO` will lead to a `SanityError`. If you don't want that, we should change the regex. But: I'd prefer for it to be the repo name (and not the full `/cvmfs/software.eessi.io` path) if it _is_ set.

Why is it useful to distinguish in the logged data between an unset and empty `EESSI_CVMFS_REPO`? Both imply the use of a local software stack anyway...

Reply:
you're right, `extractsingle` gives an error if there is no match.
`extractall` doesn't give an error but an empty list, so we can use that to avoid the try-except block:
```suggestion
        repo_name = sn.extractall(r'EESSI_CVMFS_REPO: /cvmfs/(?P<repo>.*)$', f'{self.stagedir}/{self.stdout}', 'repo', str)
        if repo_name:
                self.cvmfs_repo_name = repo_name[0]
```